ffc/process_charge_file.py

```diff
@@ -0,0 +1,129 @@
+
+SEARCH_COLUMN_NAME = "Subscription Search Value"
+
+
+async def read_the_response_in_chunks_and_write_to_file(
+    response: httpx.Response, file_path, chunk_size=1024
+):
+    """
+    This function reads the response in chunks and writes it to a file asynchronously.
+    """
+    async with aiofiles.open(file_path, "wb") as file:
+        async for chunk in response.aiter_bytes(chunk_size=chunk_size):
+            await file.write(chunk)
+    logger.info(f"Charge File Downloaded and saved: {file_path}")
+    return file_path
+
+
+class ChargeFileSplitProcessor:
+    def __init__(self, mtp_base_api_url: str):
+        self.finops_client = get_httpx_ffc_api_client()
+        self.MPT_API_BASE_URL = mtp_base_api_url
+
+    async def fetch_generated_charge_files(self) -> list:
+        """
+        This method fetches all the charge files in GENERATED status
+        """
+        return await self.finops_client.get_generated_charge_files()
+
+    async def download_zip_charge_file_from_azure_and_store(
+        self, client: httpx.AsyncClient, charge_file: dict[str, Any], tmp_dir: str
+    ) -> str | None:
+        """
+        This method downloads asynchronously downloads a charge file from the FFC_OPERATIONS API
+        and stores it as a zip file in a temporary directory.
+        Args:
+            client (httpx.AsyncClient): The HTTP client used to make asynchronous requests.
+            charge_file (dict[str, Any]): A dictionary containing the charge file.
+            tmp_dir: The temporary directory to store the downloaded zip file.
+        """
+        charge_file_id = charge_file["id"]
+        initial_url = f"{self.finops_client.api_base_url}/ops/v1/charges/{charge_file_id}/download"
+        headers = self.finops_client.get_ffc_operations_authorization_headers()
+        try:
+            response = await client.get(initial_url, headers=headers)
+            if response.status_code != 307:
+                logger.error(
+                    f"Unexpected status code {response.status_code} for {charge_file_id}"
+                )
+                return None
+            redirect_url = response.headers[
+                "Location"
+            ]  # extract the download url location from the headers
+            url_response = await client.get(
+                redirect_url
+            )  # the URL has a short-lived sas token
+            if url_response.status_code != 200:
+                logger.error(
+                    f"Failed to download  {charge_file_id}, status: {url_response.status_code}"
+                )
+                return None
+            file_path = os.path.join(tmp_dir, f"{charge_file_id}.zip")
+            return await read_the_response_in_chunks_and_write_to_file(
+                url_response, file_path
+            )
+        except Exception as error:
+            logger.exception(f"Error downloading {charge_file_id}: {error}")
+            return None
+
+    async def process_generated_charge_files(
```

Comment:
This function is not processing the charges file so I would suggest either join this function with `download_zip_charge_file_from_azure_and_store` either rename the `download_zip_charge_file_from_azure_and_store` to `get_charges_file_download_url` and rename the `process_generated_charge_files` to `download_charges_file`